src/worker/core/trade/makeItWork.ts

```diff
@@ -117,6 +117,7 @@ const tryAddAsset = async (
 		return;
 	}
 
+	const addAssetKey = Math.random();
```

Comment:
Does this make sense?

If we only recompute the pick rank when this `addAssetKey` changes, then it'll be the same rank for every value in the loop below, which is evaluating it for many different players/picks in the trade. So it'll calculate it for the first one, and then be stale for all the others.

Instead, it might make more sense to compute the cache key inside `getModifiedPickRank`. It could be something like `JSON.stringify([tid, gp, pids])` - so it would be reset any time a game is played or the team/players in the trade change.

But then again, isn't that going to be quite frequent? Each time `valueChange` is called again for the same teams in a loop, you change the assets in the trade. Sometimes you're only changing picks or the user's players, but probably like 30-40% of the time it's changing a player on the other team, which invalidates the cache. So does this cache even matter?

Reply:
So if you notice in line 148, the addAssetKey is used only when a draft pick is being added to the existing assets in this add asset forloop.

When a draft pick is added to a prospective trade (sometimes 7-8 draft picks are evaluated in a row), none of the players in the trade change, so rather than re-calculating the same team ovr 7-8 times in a row for different draft picks, it just re-uses the first projected pick that was calculated

Reply:
I noticed this behavior in previous PRs I made, where the same set of player ovrs where being used to calculated team ovr over and over again.

I can remove this caching if you think its unneeded complexity

Reply:
I missed that, my bad!

I think the biggest help the cache could ever make is when asking for offers on the trading block in baseball. In that case, it is like 25% faster with the cache. I think that's not enough to make the code more complex... trade code is already too complex! The other cache I only added when I put AI-to-AI trades in the game, because it slowed down normal gameplay. So please get rid of that cache an make this PR even smaller :)

Reply:
Sounds good! I'll remove it